torch/fx/graph.py

```diff
@@ -1801,6 +1803,18 @@ def has_side_effect(node):
                 self.erase_node(node)
                 changed = True
 
+        # Call DCE on the subgraphs
+        if self.owning_module is not None:
+            subgraph_names: OrderedSet[str] = OrderedSet()
+            for node in self.find_nodes(op="get_attr"):
+                attr_name = node.target
+                if "." not in attr_name and attr_name not in subgraph_names:
+                    sub_mod = getattr(self.owning_module, attr_name)
+                    if isinstance(sub_mod, torch.fx.GraphModule):
+                        subgraph_names.add(attr_name)
+                        changed |= sub_mod.graph.eliminate_dead_code()
+                        sub_mod.recompile()
```

Comment:
one last callout - since you are changing the generic `torch.fx.graph.eliminate_dead_code()` to effectively run on more code than it used to, there is probably some small risk that users of `torch.fx` outside of compile were (accidentally) relying on DCE not running on inner graphs.

This PR seems like the right thing to do anyway, but if you want to make revert a bit less likely you could add a config somewhere to turn off the recursing-on-subgraphs behavior. I think it's up to you.